src/main/java/com/powsybl/network/hypothesis/HypothesisUtils.java

```diff
@@ -0,0 +1,133 @@
+        LineAdder adder1 = network.newLine()
+                .setId(line.getId() + "_1")
+                .setEnsureIdUnicity(true)
+                .setVoltageLevel1(line.getTerminal1().getVoltageLevel().getId())
+                .setVoltageLevel2(voltageLevel.getId())
+                .setR(line.getR() * rdp)
+                .setX(line.getX() * xdp)
+                .setG1(line.getG1() * g1dp)
+                .setB1(line.getB1() * b1dp)
+                .setG2(line.getG2() * g2dp)
+                .setB2(line.getB2() * b2dp);
+        LineAdder adder2 = network.newLine()
+                .setId(line.getId() + "_2")
+                .setEnsureIdUnicity(true)
+                .setVoltageLevel1(voltageLevel.getId())
+                .setVoltageLevel2(line.getTerminal2().getVoltageLevel().getId())
+                .setR(line.getR() * (1 - rdp))
+                .setX(line.getX() * (1 - xdp))
+                .setG1(line.getG1() * (1 - g1dp))
+                .setB1(line.getB1() * (1 - b1dp))
+                .setG2(line.getG2() * (1 - g2dp))
+                .setB2(line.getB2() * (1 - b2dp));
+        attachLine(line.getTerminal1(), adder1, (bus, adder) -> adder.setConnectableBus1(bus.getId()), (bus, adder) -> adder.setBus1(bus.getId()), (node, adder) -> adder.setNode1(node));
+        attachLine(line.getTerminal2(), adder2, (bus, adder) -> adder.setConnectableBus2(bus.getId()), (bus, adder) -> adder.setBus2(bus.getId()), (node, adder) -> adder.setNode2(node));
+        if (topologyKind == TopologyKind.BUS_BREAKER) {
+            Bus bus = voltageLevel.getBusBreakerView()
+                    .newBus()
+                    .setId(line.getId() + "_BUS")
+                    .setEnsureIdUnicity(true)
+                    .setFictitious(true)
+                    .add();
+            adder1.setBus2(bus.getId());
+            adder2.setBus1(bus.getId());
+        } else if (topologyKind == TopologyKind.NODE_BREAKER) {
+            voltageLevel.getNodeBreakerView()
+                    .newInternalConnection()
+                    .setNode1(0)
+                    .setNode2(1)
+                    .add();
+            adder1.setNode2(0);
+            adder2.setNode1(1);
+        } else {
+            throw new AssertionError();
+        }
+        line.remove();
+        adder1.add();
+        adder2.add();
+    }
+
+    private static void attachLine(Terminal terminal, LineAdder adder, BiConsumer<Bus, LineAdder> connectableBusSetter,
+                                   BiConsumer<Bus, LineAdder> busSetter, BiConsumer<Integer, LineAdder> nodeSetter) {
+        if (terminal.getVoltageLevel().getTopologyKind() == TopologyKind.BUS_BREAKER) {
+            connectableBusSetter.accept(terminal.getBusBreakerView().getConnectableBus(), adder);
+            Bus bus = terminal.getBusBreakerView().getBus();
+            if (bus != null) {
+                busSetter.accept(bus, adder);
+                adder.setBus1(bus.getId());
+            }
```

Comment:
else throw AssertionError?

Reply:
No it is not an error if the bus is `null`, the line is just not connected (which happens)

Reply:
oh, yes I missed that